collated/main/vivekscl.md

```diff
@@ -164,6 +172,168 @@
+            break;
+        }
+
+        return height;
+    }
+
+    /**
+     * Returns true if given windowSize is a valid window size.
+     */
+    public static boolean isValidWindowSize(String windowSize) {
+        boolean isSmallWindowSize = windowSize.equalsIgnoreCase(SMALL_WINDOW_SIZE_INPUT);
+        boolean isMediumWindowSize = windowSize.equalsIgnoreCase(MEDIUM_WINDOW_SIZE_INPUT);
+        boolean isBigWindowSize = windowSize.equalsIgnoreCase(BIG_WINDOW_SIZE_INPUT);
+        return isSmallWindowSize || isMediumWindowSize || isBigWindowSize;
+    }
+
+
+}
+```
+###### /java/seedu/address/model/AddressBook.java
+``` java
+    /**
+     * Adds a person to the address book.
+     * Also checks the new person's tags and updates {@link #tags} with any new tags found,
+     * and updates the Tag objects in the person to point to those in {@link #tags}.
+     *
+     * @throws DuplicatePersonException if an equivalent person already exists.
+     */
+    public void addPerson(ReadOnlyPerson p) throws DuplicatePersonException {
+        Person newPerson = new Person(p);
+        try {
+            persons.add(newPerson);
+            syncMasterTagListWith(newPerson);
+        } catch (DuplicatePersonException e) {
+            throw new DuplicatePersonException();
+        }
+    }
+
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedReadOnlyPerson}.
+     * {@code AddressBook}'s tag list will be updated with the tags of {@code editedReadOnlyPerson}.
+     *
+     * @throws DuplicatePersonException if updating the person's details causes the person to be equivalent to
+     *      another existing person in the list.
+     * @throws PersonNotFoundException if {@code target} could not be found in the list.
+     *
+     * @see #syncMasterTagListWith(Person)
+     */
+    public void updatePerson(ReadOnlyPerson target, ReadOnlyPerson editedReadOnlyPerson)
+            throws DuplicatePersonException, PersonNotFoundException {
+        requireNonNull(editedReadOnlyPerson);
+
+        Person editedPerson = new Person(editedReadOnlyPerson);
+        try {
+            persons.setPerson(target, editedPerson);
+            syncMasterTagListWith(editedPerson);
+        } catch (DuplicatePersonException e) {
+            throw new DuplicatePersonException();
+        } catch (PersonNotFoundException e) {
+            throw new PersonNotFoundException();
+        }
+
+    }
+
+```
+###### /java/seedu/address/model/AddressBook.java
+``` java
+    @Override
+    public String toString() {
+        Iterator<Person> personIterator = persons.iterator();
+        Iterator<Tag> tagIterator = tags.iterator();
+        String allPersonsString = "Persons: ";
+        String allTagsString = "Tags: ";
+
+        if (personIterator.hasNext()) {
+            allPersonsString += "\n" + personIterator.next() + "\n";
+        }
+        if (tagIterator.hasNext()) {
```

Comment:
spacing

Reply:
Spacing as in, add a space between the if statements?